kani-compiler/src/codegen_cprover_gotoc/codegen/function.rs

```diff
@@ -286,4 +292,11 @@ pub mod rustc_smir {
             CovTerm::Zero
         }
     }
+
+    pub fn span_file_name(tcx: TyCtxt<'_>, span: Span) -> Symbol {
```

Comment:
Instead of copying this over from rustc, could you try converting `body_span` into a StableMIR span and then call `get_filename`? I think we should only paste a private function from rustc into Kani as a last resort.

Reply:
Updated in the latest commit.